tests/config/test_config_pyproject_toml.py

```diff
@@ -0,0 +1,70 @@
+from pathlib import Path
+from typing import List
+from unittest.mock import patch
+
+import pytest
+
+from pylint import run_pylint
+
+HERE = Path(__file__).parent
+
+
+@pytest.mark.parametrize(
+    "pyproject_toml_path,expected_results",
+    [
+        [
+            "issue_4580",
+            {
+                "basic_example.toml": {"code": 0},
+                "correct_basic_name_group.toml": {"code": 0},
+                "correct_import_preferred_module.toml": {"code": 0},
+                "empty_list.toml": {
+                    "code": 2,
+                    "out": "encountered 'load-plugins' : '[]'",
+                },
+                "invalid_data_for_basic.toml": {
+                    "code": 2,
+                    "out": "encountered 'load-plugins' : '[]'",
+                },
+                "invalid_data_for_import.toml": {"code": 0},
+                "rich_types.toml": {"code": 0},
+                "top_level_disable.toml": {
+                    "code": 2,
+                    "out": "encountered 'disable' : 'logging-not-lazy,",
+                },
+            },
+        ],
+        [
+            "issue_4746",
+            {
+                "loaded_plugin_does_not_exists.toml": {
+                    "code": 2,
+                    "out": "'pylint_websockets' is impossible to load",
+                }
+            },
+        ],
+    ],
+)
+def test_config_pyproject_toml(pyproject_toml_path, expected_results, capsys):
+    pyproject_toml_paths: List[Path] = (HERE / pyproject_toml_path).iterdir()
+    for toml_path in pyproject_toml_paths:
+        expected_result = expected_results[toml_path.name]
+        file_to_lint = str(HERE / "file_to_lint.py")
+        with patch(
+            "sys.argv",
+            ["pylint", "--rcfile", str(toml_path), file_to_lint, "-r", "n"],
+        ):
+            try:
+                run_pylint()
+            except SystemExit as ex:
```

Comment:
Not parametrising this test might create a lot of duplicate lines but these asserts are really hard to follow. 
Wouldn't it be better to just separate these into different tests and assert on the value of the exit code instead of parametrising?

Reply:
I'm half inclined to create a file with expected result like for the functional tests.

Reply:
Wouldn't we need to create a new system for it? 
With the functional tests you can still see the expected behaviour (because of the comments) in the tests, while the full data is in another file. That's quite useful.

Reply:
This is going to require another merge request I guess. Do you think we should do this before or after we handle this one ? I'm going to push what I currently have (without annotation in the toml for expected error).

Reply:
I'm not sure how easy it is to implement such a system. Might be a bit overkill?

Separating the tests might be easier? It is not as if we're going for the most perfomance-optimized test suite anyway. It might take a similar amount of time as creating a new system to allow external expected results.